app/routes/app._index.tsx

```diff
@@ -92,6 +80,29 @@ export const action = async ({ request }: ActionFunctionArgs) => {
 };
 
 export default function Index() {
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const sLinkRemixConnector = (event: MouseEvent) => {
+      if (
+        (event.target as HTMLElement)?.tagName !== "S-LINK" ||
+        !(event.target as HTMLAnchorElement).href ||
+        (event.target as HTMLAnchorElement).target === "_blank"
+      )
+        return;
+
+      event.preventDefault();
+      const url = (event.target as HTMLAnchorElement).href;
+      navigate(url);
+    };
+
+    document.body.addEventListener("click", sLinkRemixConnector);
+
+    return () => {
+      document.body.removeEventListener("click", sLinkRemixConnector);
+    };
+  }, [navigate]);
```

Comment:
@developit quick shim @gwyneplaine and I worked on. Is this what you were thinking? I was unsure where this should live.